pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -118,65 +140,346 @@ func (bm *backupManager) Sync(backup *v1alpha1.NebulaBackup) error {
 	return utilerrors.ReconcileErrorf("waiting for backup job [%s/%s] done", backup.Namespace, backup.Name)
 }
 
-func generateBackupJob(backup *v1alpha1.NebulaBackup, metaAddr string) *batchv1.Job {
-	bpCmdHead := "exec /usr/local/bin/br-ent backup full"
-	if backup.Spec.BR.BackupName != "" {
-		bpCmdHead = fmt.Sprintf("exec /usr/local/bin/br-ent backup incr --base %s", backup.Spec.BR.BackupName)
+func getScheduledBackupScriptsObjectKey(backup *v1alpha1.NebulaBackup) client.ObjectKey {
+	return client.ObjectKey{
+		Namespace: backup.Namespace,
+		Name:      fmt.Sprintf("backup-scripts-cm-%v", backup.Name),
 	}
+}
 
-	backupCmd := fmt.Sprintf("%s --meta %s --storage s3://%s --s3.access_key $%s --s3.secret_key $%s --s3.region %s --s3.endpoint %s",
-		bpCmdHead, metaAddr, backup.Spec.BR.S3.Bucket, EnvS3AccessKeyName, EnvS3SecretKeyName, backup.Spec.BR.S3.Region, backup.Spec.BR.S3.Endpoint)
+func generateScheduledBackupScriptsConfigMap(backup *v1alpha1.NebulaBackup) *corev1.ConfigMap {
+	nbObjectKey := getScheduledBackupScriptsObjectKey(backup)
 
-	return &batchv1.Job{
+	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      backup.Name,
-			Namespace: backup.Namespace,
+			Name:      nbObjectKey.Name,
+			Namespace: nbObjectKey.Namespace,
+			Labels: map[string]string{
+				"backup-name":      backup.Name,
+				"backup-namespace": backup.Namespace,
+			},
 		},
-		Spec: batchv1.JobSpec{
-			Parallelism:  pointer.Int32(1),
-			Completions:  pointer.Int32(1),
-			BackoffLimit: pointer.Int32(0),
-			Template: corev1.PodTemplateSpec{
-				Spec: corev1.PodSpec{
-					ImagePullSecrets: backup.Spec.ImagePullSecrets,
-					Containers: []corev1.Container{
-						{
-							Name:            "backup",
-							Image:           backup.Spec.ToolImage,
-							ImagePullPolicy: backup.Spec.ImagePullPolicy,
-							Env: []corev1.EnvVar{
-								{
-									Name: EnvS3AccessKeyName,
-									ValueFrom: &corev1.EnvVarSource{
-										SecretKeyRef: &corev1.SecretKeySelector{
-											LocalObjectReference: corev1.LocalObjectReference{
-												Name: backup.Spec.BR.S3.SecretName,
+		Data: map[string]string{ // TODO: remove x. x means no need to configure, but it is a required field in the program.
+			"backup-cleanup.sh": `#!/bin/bash
+ACCESS_KEY=$S3_ACCESS_KEY
+SECRET_KEY=$S3_SECRET_KEY
+
+function usage {
+    echo "usage $0 [-h] < -m meta_address, -s storage, -r s3.region, -e s3.endpoint, -b max_backups, -t max_reserved_time >"
+    echo "-h displays this message"
+    echo "-m specifies the meta server address"
+    echo "-s specifies the backup target url"
+    echo "-r specifies the s3 bucket region"
+    echo "-e specifies the s3 endpoint"
+    echo "-b specifies the maximum number of backups"
+    echo "-t spefifies the maximum reserved time for backups"
+}
+
+function convert_to_sec {
```

Comment:
what if the input of "10M20d"

Reply:
@MegaByte875 Good catch. Didn't consider this case so it'll currently return invalid input. Will fix this logic by calculating the earliest time a backup should be kept in the NebulaScheduleBackup reconciler and passing that to this script.

Reply:
Fixed